### DISCUSSION TITLE
strange perturbations in steady state time derivative solution

### DISCUSSION THREAD

**Discussion by swonner** [idaholab/moose#29560] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am investigating the following simple system equations:

${dX_V\over dt} = K_0$

${dX_I\over dt} = K_0$

These terms are a small part of a more complex system I am trying to inevitably solve. I attached this system at the bottom of this post for reference. 

When I solve the above system and plot the concentration profile across my mesh, I have noticed that there are strange perturbations near the sides of the mesh. These perturbations seem to begin right at the first timestep and remain over time. The concentration does increase as a function of K0 over time, however this profile is puzzling because the expected result was for the concentrations (XV and XI) to be constant across the entire mesh. See the attached line scan picture for the first timestep, and input file below.

Could someone please advise how I might correct this behavior? 

**line scan across the mesh at time step 1**
![image](https://github.com/user-attachments/assets/b7a2cf51-8d3f-4ac1-a1b6-84974e5719ac)

**Input File**
```
[Mesh] 
    [./gen]
        type = GeneratedMeshGenerator
        dim = 2
        nx = 50
        ny = 50
        xmax = 50 # nm
        ymax = 50 # nm
    [../]
[]

[Variables] 
    [./XV]
        order = FIRST
        family = LAGRANGE
    [../]
    [./XI]
        order = FIRST
        family = LAGRANGE
    [../]  
[]

[ICs]
    [./XV_ic]
        type = ConstantIC
        variable = XV
        value = 1e-15 
    [../]
    [./XI_ic]
        type = ConstantIC
        variable = XI
        value = 1e-15
    [../]
[]

[Kernels]
    #################### Main Variable: XV ####################
    ## (𝜕𝑋𝑉)/𝜕𝑡 - 𝐾0 = 0
   
    [./timeXV] #(𝜕𝑋𝑉)/𝜕𝑡
        type = ADTimeDerivative
        variable = XV
    [../]
    [./defectgenV] #- 𝐾0
        type = ADMatBodyForce
        variable = XV
        material_property = K0 
    [../]

    #################### Main Variable: XI ####################
    ## (𝜕𝑋𝐼)/𝜕𝑡 -  𝐾0  = 0

    [./timeXI] #(𝜕𝑋𝐼)/𝜕𝑡
        type = ADTimeDerivative
        variable = XI
    [../]
    [./defectgenI] #- 𝐾0
        type = ADMatBodyForce
        variable = XI
        material_property = K0 
    [../]

[]


[Materials]

    [./SettingGeneralConstants]
        type = ADGenericConstantMaterial
        prop_names = 'K0'
        prop_values = '1.00e-05’
    [../]
       
[]


[Preconditioning]
    [my_prec]
      type = SMP
      full = true
    []
[]   


[Executioner]
  type = Transient
  solve_type = NEWTON
  l_max_its = 20 
  l_tol = 1e-12 
  nl_max_its = 15 
  nl_rel_tol = 1e-12 
  nl_abs_tol = 1e-12 
  end_time = 10  
  
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = ' hypre    boomeramg'

  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 1  # The default timestep size between solves
    cutback_factor = 0.5 
    growth_factor = 1.2 
    optimal_iterations = 7 
  [../]
[]


[Outputs]
    file_base = model
    [./exodus]
        type = Exodus
        print_linear_residuals = true
    [../]
[]

[Debug]
    show_var_residual_norms = true
[]

```

**The more complex system I would like to eventually solve** 
This system relies on &nabla;X. Therefore, it is important that I remove the strange perturbations in concentration that is incorrectly being introduced above so that I can achieve a correct spatially resolved solution for this system. 
![image](https://github.com/user-attachments/assets/aec02501-d5a4-4330-9afe-57a5b5662277)

**Comment by GiudGiud** [comment]
Hello

Can you rescale the plot?

It looks to me like this is just numerical precision error of a constant solution. It would be good to see the amplitude of these oscillations to decide if they are relevant

**Reply by swonner** [reply]
Hi GiudGiud, thank you for looking into my question!

I have replotted and scaled the plot. 
![image](https://github.com/user-attachments/assets/da4eab81-02c2-49bc-b512-229b1e0a5a51)

**Reply by GiudGiud** [reply, marked as answer]
Ok these oscillations are likely irrelevant and nothing to worry about. They are to the order of 1e-16 or numerical precision (double precision by default in moose)

**Reply by swonner** [reply]
Thank you!
